mission_control/navigator_motion_planner/nodes/explorer.py

```diff
@@ -0,0 +1,137 @@
+        origin = navigator_tools.pose_to_numpy(self.map_metadata.origin)[0]
+
+        # Convert from meters to grid measurements
+        center_grid = (cpm * (position - origin)).astype(np.int64)
+        radius_grid = int(radius * cpm)
+
+        cv2.circle(self.grid, tuple(center_grid[:2]), radius, 1, -1)
+
+    def _lidar_view(self, pose):
+        radius = 30  # m
+        position, quat = pose
+        yaw = tf.transformations.euler_from_quaternion([quat[0], quat[1], quat[2], quat[3]])[2]
+
+        # Need to get the position on the ogrid to put the shape
+        cpm = 1 / self.map_metadata.resolution  # cells per meter
+        origin = navigator_tools.pose_to_numpy(self.map_metadata.origin)[0]
+
+        # Convert from meters to grid measurements
+        center_grid = (cpm * (position - origin - radius)).astype(np.int64)[:2]
+        radius_grid = int(radius * cpm)
+
+        # Draw the shape on it's own canvas then combine
+        shape = np.zeros((2 * radius_grid, 2 * radius_grid))  # Blank place to put the cone
+        cv2.circle(shape, (radius_grid, radius_grid), radius_grid, 1, -1)
+
+        # Get rid of that triangle of unseen space
+        pts = np.array([[radius_grid, radius_grid], shape.shape, [0, shape.shape[1]]])
+        pts = pts.reshape((-1,1,2)).astype(np.int32)
+        cv2.fillPoly(shape, [pts], 0)
+
+        # Then rotate so that we are oriented the same as the boat
+        M = cv2.getRotationMatrix2D((radius_grid, radius_grid), np.degrees(yaw) - 90, 1.0)
+        shape = cv2.warpAffine(shape, M, shape.shape[:2][::-1])
+
+        # Add the shape to the grid (you have to do some funky flipping here)
+        self.grid[center_grid[1]:center_grid[1] + int(2 * radius_grid),
+                  center_grid[0]:center_grid[0] + int(2 * radius_grid)] += np.flipud(shape)
+
+    def reset_grid(self):
+        print "Resetting Grid."
+        self.grid = np.zeros((self.map_metadata.height, self.map_metadata.width)) - 1
+
+    def pub_ogrid(self, *args):
+        ogrid = OccupancyGrid()
+        ogrid.header = navigator_tools.make_header(frame="enu")
+        ogrid.info = self.map_metadata
+
+        # Over saturate and clip to get range of [-1, 1]
+        self.grid = np.clip(self.grid * 1000, -1, 1)
+        ogrid.data = self.grid.flatten().astype(np.int8).tolist()
+
+        self.ogrid_pub.publish(ogrid)
+
+class Explorer(object):
+    def __init__(self):
+        self.navigator = Navig
+
+        #self.db = DatabaseInterface()
+
+        self.ogrid = OGridManager()
+        self.ogrid.draw_shape = self.ogrid._lidar_view
+
+        self.has_control = False
+        self.move_result = None
+
+        rospy.Subscriber("/odom", Odometry, self.got_odom)
+        rospy.Service("/explorer/explore", SetBool, self.search_request)
+
+        #while self._nav_dummy.pose is None and not rospy.is_shutdown():
+        #    rospy.sleep(.1)
+
+        print "Taking control"
```

Comment:
Doesn't say what's taking control. If it was only for debugging it should be removed.